src/Common/StringSearcher.h

```diff
@@ -569,8 +569,10 @@ class StringSearcher<true, ASCII> : public StringSearcherBase
     uint8_t first{};
 
 #ifdef __SSE4_1__
-    /// vector filled `first` for determining leftmost position of the first symbol
-    __m128i pattern;
+    uint8_t second{};
```

Comment:
About `#ifdef __SSE4_1__`: As far as I see (--> [Intel Intrinsics Guide](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html)), all used intrinsics in the block l. 593ff and the block l. 671 are SSE2 intrinsic. Could we ifdef for SSE2 instead?

Reply:
Done.

Reply:
The same comment also applies to all other code blocks touched in this PR.

Reply:
Yes, you're right. I have checked all the intrinsic code part. "_mm_insert_epi8" is from SSE4.1 and all the others are from SSE2. I have pushed a patch to correct it. But I have a question to do that.

The platform from SSE2 to SSE4.1 could not use the SSE4.1 acceleration. The SSE2 part code is useless and waste time for the platform from SSE2 to SSE4.1. All the code block work together for the same goal no matter from the SSE2 or the SSE4.1.
So we might align all the code blocks to SSE4.1?

Reply:
Good point, I seemingly missed `_mm_insert_epi8`. In that case, yes, you could align all the code to SSE4.1. It actually doesn't matter since the ClickHouse build itself requires at least SSE 4.2 (file "cmake/cpu_features.cmake")...

Reply:
If the platform is SSE3, then it will execute all the SSE2 code block. But it still could not use the SIMD acceleration as it need the SSE4.1 code block. And it will fall back to the "Byte-by-Byte comparison". Then the SSE2 code block might waste time for the SSE3 platform?

Reply:
OK. I have aligned all the code block to SSE4.1.